src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala

```diff
@@ -12,87 +12,186 @@ trait ClassLikeSupport:
   self: TastyParser =>
   import reflect._
 
-  def parseClass(classDef: reflect.ClassDef)(using ctx: Context): DClasslike =
-    val parents = for
-      parentTree <- classDef.parents if isValidPos(parentTree.pos)  // We assume here that order is correct
-      parentSymbol = if (parentTree.symbol.isClassConstructor) parentTree.symbol.owner else parentTree.symbol 
-        if parentSymbol != defn.ObjectClass
-    yield parentTree.dokkaType
+  extension (c: ClassDef):
+      def getExtensionGroups: List[ExtensionGroup] = {
+          val extensions = c.symbol.classMethods.filterNot(isSyntheticFunc).filter(isExtensionMethod(_))
 
-    val (extensions, methods) = classDef.symbol.classMethods.filterNot(isSyntheticFunc).partition(isExtensionMethod(_))
+          extensions.map(e => (getExtendedSymbol(e).get, e) ).groupBy{
+            case (arg, e) => arg.symbol.pos
+          }.map{
+            case (pos, list) => if(list.size == 1) {
```

Comment:
```
case class ExtensionPosition(pos: Position)(parameter: DParameter) // only first list is taken into acount for quals etc.
val grouppeed = extension.groupBy(  e => 
  val extensionArg = getExtendedSymbol(e).get
  ExtensionPosition(extensionArg.symbol.position)(extensionArg)
}
grouppeed.map { 
  case (extension, methods) =>
      val isGroupped = methods.size > 1
      val dMethods = methods.map(m => parseMethod(m, extInfo = Some(ExtensionInformation(isGroupped))))
      ExtensionGroup(parseArgument(extension.argument, _ => "", isExtendedSymbol = true, isGroupped), dMethods)
}
```

Reply:
did sth like that, but there was problem with equality of extensionArgs